src/components/item/sharing/ManageShortLink.tsx

```diff
@@ -0,0 +1,256 @@
+        alias: initialAlias!, // old alias name use to find the shortLink to patch
+        shortLink: {
+          alias: shortLink,
+          platform: linkType,
+        },
+      });
+    }
+
+    setSaved(true);
+    setDisableSave(true);
+    handleClose();
+  };
+
+  const onClose = (_event: Event, reason: string) => {
+    if (reason === 'backdropClick') {
+      return;
+    }
+
+    handleClose();
+  };
+
+  const WrapHelper = styled(FormHelperText)(() => ({
+    overflowWrap: 'anywhere',
+    whiteSpace: 'normal',
+    maxWidth: '240px',
+  }));
+
+  return (
+    <Dialog
+      open={open}
+      onClose={onClose}
+      aria-labelledby="alert-dialog-title"
+      aria-describedby="alert-dialog-description"
+    >
+      <DialogTitle id="alert-dialog-title">{DIALOG_TITLE}</DialogTitle>
+      <DialogContent>
+        <Stack direction="column" alignItems="left" spacing={2} mt={2}>
+          <Stack direction="row" alignItems="center">
+            <TextField
+              id={SHORT_LINK_ALIAS_INPUT_ID}
+              label="Alias"
+              variant="outlined"
+              size="small"
+              value={shortLink}
+              onChange={handleAliasChange}
+              error={hasError()}
+            />
+            <Tooltip title={translateBuilder(BUILDER.GENERATE_ALIAS_TOOLTIP)}>
+              <span>
+                <IconButton
+                  onClick={() => setShortLink(randomAlias())}
+                  id={SHORT_LINK_RANDOMIZE_BUTTON_ID}
+                >
+                  <CachedIcon />
+                </IconButton>
+              </span>
+            </Tooltip>
+          </Stack>
+
+          {hasError() && <WrapHelper error>{aliasError}</WrapHelper>}
+          {!hasError() && <WrapHelper>{aliasMsg}</WrapHelper>}
+
+          <Select
+            sx={{ ml: 1, textTransform: 'capitalize' }}
+            value={linkType}
+            onChange={handleLinkTypeChange}
+            renderValue={(value) => enumT(value)}
+            id={SHORT_LINK_PLATFORM_SELECT_ID}
+            size="small"
+          >
+            <MenuItem value={Context.Builder}>
+              {enumT(Context.Builder)}
+            </MenuItem>
+            <MenuItem value={Context.Player}>{enumT(Context.Player)}</MenuItem>
+            <MenuItem value={Context.Library}>
+              {enumT(Context.Library)}
+            </MenuItem>
+          </Select>
+        </Stack>
+      </DialogContent>
+      <DialogActions>
+        <Button
+          onClick={handleClose}
+          autoFocus
+          id={buildShortLinkCancelBtnId(shortLink)}
+        >
+          {translateBuilder(BUILDER.CANCEL_BTN)}
+        </Button>
+        <Button
+          color="success"
+          onClick={handleSaveAlias}
+          disabled={disableSave || hasError()}
+          id={SHORT_LINK_SAVE_BUTTON_ID}
+        >
+          {translateBuilder(BUILDER.SAVE_BTN)}
+        </Button>
+      </DialogActions>
+    </Dialog>
+  );
+};
```

Comment:
It's a quite long file if you can split it in more files/components it would be better.